ompi/mpi/fortran/base/gen-mpi-mangling.pl

```diff
@@ -146,9 +165,19 @@ sub gen_c_constants_decl {
     foreach my $key (sort(keys(%{$fortran}))) {
         my $f = $fortran->{$key};
         my $m = mangle($f->{c_name});
-        print OUT "extern $f->{c_type} $m;
-#define OMPI_IS_FORTRAN_" . uc($key) . "(addr) \\
-        (addr == (void*) &$m)\n\n";
+        my @all_names = mangle_all($f->{c_name});
+        foreach my $n (@all_names) {
+            print OUT "extern $f->{c_type} $n;\n";
+        }
+        print OUT "#define OMPI_IS_FORTRAN_" . uc($key) . "(addr) \\
+        (";
+        for( my $i = 0; $i < scalar(@all_names); ++$i ) {
+            print OUT "addr == (void*) &".$all_names[$i];
+            if( scalar(@all_names)-1 != $i ) {
+                print OUT " || \\\n         ";
```

Comment:
Note that we intentionally did not do this because it means that every comparison to a sentinel Fortran value means that there will now be up to 4 comparisons.  This is a potential/probable performance penalty for users who only have a single Fortran compiler.  

That being said, I don't know if modern compilers/processors are able to mitigate the cost of the 4 comparisons such that the performance difference would be negligible.  Have you done any performance testing on POWER and x86-64?

**Sidenote:** We did all 4 weak symbols (where supported) because there's no additional performance penalty for this, but doing all 4 variations of the Fortran constant here could add latency.